docs/server-config.md

```diff
@@ -20,31 +22,132 @@
+
+```sh
+tabpy --config=path/to/my/config/file.conf
+```
 
 The default config file is provided to show you the default values but does not
 need to be present to run TabPy.
 
+### Configuration File Content
+
+Configuration file consists of settings for TabPy itself and Python logger
+settings. You should only set parameters if you need different values than
+the defaults.
+
+TabPy parameters explained below, the logger documentation can be found
+at [`logging.config` documentation page](https://docs.python.org/3.6/library/logging.config.html).
+
+`[TabPy]` parameters:
+
+- `TABPY_PORT` - port for TabPy to listen on. Default value - `9004`.
+- `TABPY_QUERY_OBJECT_PATH` - query objects location. Used with models, see
+  [TabPy Tools documentation](tabpy-tools.md) for details. Default value -
+  `/tmp/query_objects`.
+- `TABPY_STATE_PATH` - state location for Tornado web server. Default
+   value - `tabpy/tabpy_server` subfolder in TabPy package folder.
+- `TABPY_STATIC_PATH` - location of static files (index.html page) for
+  TabPy instance. Default value - `tabpy/tabpy_server/static` subfolder in
+  TabPy package folder.
+- `TABPY_PWD_FILE` - path to password file. Setting up this parameter
+  makes TabPy require credentials with HTTP(S) requests. More details about
+  authentication can be found in [Authentication](#authentication)
+  section. Default value - not set.
+- `TABPY_TRANSFER_PROTOCOL` - transfer protocol. Default value - `http`. If
+  set to `https` two additional parameters have to be specified:
+  `TABPY_CERTIFICATE_FILE` and `TABPY_KEY_FILE`. More details for how to
+  configure TabPy for HTTPS are at [Configuring HTTP vs HTTPS]
+  (#configuring-http-vs-https) section.
+- `TABPY_CERTIFICATE_FILE` the certificate file to run TabPy with. Only used
+  with `TABPY_TRANSFER_PROTOCOL` set to `https`. Default value - not set.
+- `TABPY_KEY_FILE` to private key file to run TabPy with. Only used
+  with `TABPY_TRANSFER_PROTOCOL` set to `https`. Default value - not set.
+- `TABPY_LOG_DETAILS` - when set to `true` additional call information
+  (caller IP, URL, client info, etc.) is logged. Default value - `false`.
+- `TABPY_EVALUATE_TIMEOUT` - script evaluation timeout in seconds. Default
+  value - `30`.
+
+### Configuration File Example
+
+**Note:** _Always use absolute paths for the configuration paths
+settings._
+
+```ini
+[TabPy]
+# TABPY_QUERY_OBJECT_PATH = /tmp/query_objects
+# TABPY_PORT = 9004
+# TABPY_STATE_PATH = <package-path>/tabpy/tabpy_server
+
+# Where static pages live
+# TABPY_STATIC_PATH = <package-path>/tabpy/tabpy_server/static
+
+# For how to configure TabPy authentication read
+# docs/server-config.md.
+# TABPY_PWD_FILE = /path/to/password/file.txt
+
+# To set up secure TabPy uncomment and modify the following lines.
+# Note only PEM-encoded x509 certificates are supported.
+# TABPY_TRANSFER_PROTOCOL = https
+# TABPY_CERTIFICATE_FILE = path/to/certificate/file.crt
```

Comment:
These paths (certificate and key file) are still relative. Relative to what?

Reply:
Updated documentation and the config.